test/CalculatorW3GTest.cpp

```diff
@@ -0,0 +1,144 @@
+ * 
+ */
+TEST(CalculatorW3Tests, WhenDivideByZeroDivideCalculatorIsConstructed_ThenErrorPrintedToStdError)
+{
+    testing::internal::CaptureStderr();
+    DivideCalculatorW3 divideCalculator = DivideCalculatorW3(10, 0);
+    std::string output = testing::internal::GetCapturedStderr();
+    ASSERT_NE(output, std::string(""));
+}
+
+/**
+ * @brief Test to make sure a fully formed equation is formed when toString is called
+ * 
+ */
+TEST_F(SimpleExpressionCalculatorW3Test, WhenToStringIsCalledOnSimpleExpressionCalculators_ThenFullFormattedEquationIsReturned)
+{
+    ASSERT_TRUE(addCalculator.toString().find('=') != std::string::npos);
+    ASSERT_TRUE(addCalculator.toString().find('+') != std::string::npos);
+    ASSERT_TRUE(subtractCalculator.toString().find('=') != std::string::npos);
+    ASSERT_TRUE(subtractCalculator.toString().find('-') != std::string::npos);
+    ASSERT_TRUE(multiplyCalculator.toString().find('=') != std::string::npos);
+    ASSERT_TRUE(multiplyCalculator.toString().find('*') != std::string::npos);
+    ASSERT_TRUE(divideCalculator.toString().find('=') != std::string::npos);
+    ASSERT_TRUE(divideCalculator.toString().find('/') != std::string::npos);
+}
+
+/**
+ * @brief Test to make sure the math is happening as expected
+ * 
+ */
+TEST_F(SimpleExpressionCalculatorW3Test, WhenGetResultIsCalledOnSimpleExpressionCalculators_ThenAnswerIsReturned)
+{
+    ASSERT_EQ(20, addCalculator.getResult());
+    ASSERT_EQ(0, subtractCalculator.getResult());
+    ASSERT_EQ(100, multiplyCalculator.getResult());
+    ASSERT_EQ(1, divideCalculator.getResult());
+}
+
+/**
+ * @brief Test to make sure the expression is NOT a fully formed equation, i.e. it should NOT have an '='
+ * 
+ */
+TEST_F(SimpleExpressionCalculatorW3Test, WhenGetExpressionIsCalledOnSimpleExpessionCalculators_ThenExpressionIsReturned)
+{
+    ASSERT_TRUE(addCalculator.getExpression().find('=') == std::string::npos);
+    ASSERT_TRUE(addCalculator.getExpression().find('+') != std::string::npos);
+    ASSERT_TRUE(subtractCalculator.getExpression().find('=') == std::string::npos);
+    ASSERT_TRUE(subtractCalculator.getExpression().find('-') != std::string::npos);
+    ASSERT_TRUE(multiplyCalculator.getExpression().find('=') == std::string::npos);
+    ASSERT_TRUE(multiplyCalculator.getExpression().find('*') != std::string::npos);
+    ASSERT_TRUE(divideCalculator.getExpression().find('=') == std::string::npos);
+    ASSERT_TRUE(divideCalculator.getExpression().find('/') != std::string::npos);
+}
+
+/**
+ * @brief Test to make sure the calculator factory does not contruct calculator objects if the input is invalid
+ * 
+ */
+TEST_F(SimpleExpressionCalculatorW3FactoryTest, WhenCreateCalculatorIsCalledWithInvalidString_ThenNullPtrIsReturned)
+{
+    ASSERT_FALSE(simpleExpressionFactory.createCalculator(""));
+    ASSERT_FALSE(simpleExpressionFactory.createCalculator("Invalid Input"));
+    ASSERT_FALSE(simpleExpressionFactory.createCalculator("10 ? 10"));
+}
```

Comment:
Also, add unit test for:

1.  input is a special character
2. A long empty string
3. Number mixed with letters